lib/cylc/scheduler.py

```diff
@@ -248,19 +246,19 @@ def start(self):
         except KeyboardInterrupt as exc:
             try:
                 self.shutdown(str(exc))
-            except Exception:
+            except Exception as exc2:
                 # In case of exceptions in the shutdown method itself.
-                ERR.warning(traceback.format_exc())
+                LOG.exception(exc2)
                 sys.exit(1)
 
         except Exception as exc:
-            ERR.critical(traceback.format_exc())
-            ERR.error("error caught: cleaning up before exit")
+            LOG.exception(exc)
```

Comment:
Unless you've done something clever in the logging this will suppress the traceback we need to debug the exception (hence the `traceback.format_exc()` solution).

Reply:
`logging.exception` appears to print the full traceback (at least in my environment).

Reply:
So it does, `logging` does something clever.